opengrok-indexer/src/main/java/org/opengrok/indexer/history/PerforceRepository.java

```diff
@@ -61,14 +62,44 @@ public PerforceRepository() {
         ignoredFiles.add(".p4config");
     }
 
+    public static String protectPerforceFilename(String name) {
+        ArrayList<SimpleImmutableEntry<String,String>>   transmap = new ArrayList<SimpleImmutableEntry<String,String>>() {
+            {
+                /* NOTE: Must replace '%' first, or that translation will */
+                /* will affect the result of the others. */
+                add(new SimpleImmutableEntry<>("%", "%25"));
+                add(new SimpleImmutableEntry<>("#", "%23"));
+                add(new SimpleImmutableEntry<>("\\*", "%2A"));
+                add(new SimpleImmutableEntry<>("@", "%40"));
+            }
+        };
+        String t = name;   /* XXX remove? */
+        for (SimpleImmutableEntry<String,String> ent : transmap) {
+            t = t.replaceAll(ent.getKey(), ent.getValue());
+        }
+        /* Instead of iterating over a list, just force each? */
+/*
+        String t = name.replaceAll("%", "%25");
+        t = t.replaceAll("#", "%23");
+        t = t.replaceAll("\\*", "%2A");
+        t = t.replaceAll("@", "%40");
+*/
+        if (name != t) {
+            LOGGER.log(Level.INFO,
```

Comment:
this should probably be `Level.FINEST`

Reply:
Yeah, that was originally for my debugging and I thought it might come out.  But leaving it in at FINE or FINEST makes sense, so I'll change it as you suggest.